app/queries/project/subscriptions_pledged_value_query.rb

```diff
@@ -0,0 +1,17 @@
+class Project::SubscriptionsPledgedValueQuery
+  attr_reader :subscriptions
+
+  def initialize(project)
+    @subscriptions = project.subscriptions.includes(:plan).with_paid_transactions
+  end
+
+  def self.call(project)
+    new(project).call
+  end
+
+  def call
+    subscriptions.inject(0) do |pledged_value, subscription|
+      pledged_value + (subscription.transactions.length * subscription.plan.formatted_amount)
```

Comment:
You're right! I haven't noticed your query is already bringing the transactions that matter.